test/e2e_node/remote/node_e2e.go

```diff
@@ -101,23 +102,57 @@ func prependMemcgNotificationFlag(args string) string {
 	return "--kubelet-flags=--kernel-memcg-notification=true " + args
 }
 
-// updateOSSpecificKubeletFlags updates the Kubelet args with OS specific
-// settings.
-func updateOSSpecificKubeletFlags(args, host, workspace string) (string, error) {
-	output, err := SSH(host, "cat", "/etc/os-release")
+// osSpecificActions takes OS specific actions required for the node tests
+func osSpecificActions(args, host, workspace string) (string, error) {
+	output, err := getOSDistribution(host)
 	if err != nil {
 		return "", fmt.Errorf("issue detecting node's OS via node's /etc/os-release. Err: %v, Output:\n%s", err, output)
 	}
 	switch {
-	case strings.Contains(output, "ID=gci"), strings.Contains(output, "ID=cos"):
+	case strings.Contains(output, "fedora"), strings.Contains(output, "rhcos"),
+		strings.Contains(output, "centos"), strings.Contains(output, "rhel"):
+		return args, setKubeletSELinuxLabels(host, workspace)
+	case strings.Contains(output, "gci"), strings.Contains(output, "cos"):
 		args = prependMemcgNotificationFlag(args)
 		return prependCOSMounterFlag(args, host, workspace)
-	case strings.Contains(output, "ID=ubuntu"):
+	case strings.Contains(output, "ubuntu"):
 		return prependMemcgNotificationFlag(args), nil
 	}
 	return args, nil
 }
 
+// setKubeletSELinuxLabels set the appropriate SELinux labels for the
+// kubelet on Fedora CoreOS distribution
+func setKubeletSELinuxLabels(host, workspace string) error {
```

Comment:
Function name is more than only the Kubelet. Maybe drop Kubelet to `setSELinuxLabels`?